### PR TITLE
Lisp checks

### PR DESCRIPTION
**Pull request by dunn**
These are the only changes that need to be made to core for #41388. If this looks good, then once it's merged I'll start updating formulae.

### CODE REVIEW

File: Library/Homebrew/caveats.rb [hunk 1]
```diff
@@ -132,6 +133,15 @@ def app_caveats
     end
   end
 
+  def elisp_caveats
+    if keg && keg.elisp_installed?
+      <<-EOS.undent
+        Lisp files have been installed to:
```
Comment (xu-cheng) [inline Library/Homebrew/caveats.rb]: Maybe `Emacs Lisp` rather than simple `Lisp` here.

File: Library/Homebrew/formula_cellar_checks.rb [hunk 2]
```diff
@@ -173,6 +173,26 @@ def check_python_framework_links lib
     EOS
   end
 
+  def check_site_lisp(share, name)
```
Comment (xu-cheng) [inline Library/Homebrew/formula_cellar_checks.rb]: Method would be better to name as `check_emacs_lisp` or something.

File: Library/Homebrew/caveats.rb (hunk 1 above)
Comment (dunn) [inline Library/Homebrew/caveats.rb]: Yeah, good point.

**Comment by dunn** [comment]
I'm starting to worry it's too strict a requirement. I'll think about it a bit more.

**Pull request closed without merging**